lr_face/experiments.py

```diff
@@ -48,18 +49,28 @@ def get_calibration_and_test_pairs(self) -> Tuple[
         # identity into two disjoint sets and make pairs out of them.
         if isinstance(datasets, Dataset):
             test_size = self.data_config['fraction_test']
-            calibration_pairs, test_pairs = map(
-                make_pairs,
-                split_by_identity(datasets, test_size)
-            )
+            if datasets.pairs:
+                calibration_pairs, test_pairs = train_test_split(
+                    datasets.pairs, test_size=test_size)
+            else:
+                calibration_pairs, test_pairs = map(
+                    make_pairs,
+                    split_by_identity(datasets, test_size)
+                )
             return calibration_pairs, test_pairs
 
         # If `datasets` is already a tuple of `Dataset` instances, make pairs
         # for each individual dataset and return those.
         if isinstance(datasets, tuple) \
                 and len(datasets) == 2 \
                 and all(isinstance(x, Dataset) for x in datasets):
-            return datasets[0].pairs, datasets[1].pairs
+            pairs_0 = datasets[0].pairs
```

Comment:
why not calibration and test_pairs like above?

Reply:
ja, beter